cache/src/main/java/com/example/expenselogger/cache/ExpenseDatabase.kt

```diff
@@ -8,24 +8,16 @@ import com.example.expenselogger.cache.dao.ExpenseDao
 import com.example.expenselogger.cache.entity.ExpenseEntity
 
 @Database(entities = [ExpenseEntity::class], version = 1, exportSchema = false)
-public abstract class ExpenseDatabase : RoomDatabase() {
+internal abstract class ExpenseDatabase : RoomDatabase() {
 
     public abstract val expenseDao: ExpenseDao
 
     public companion object {
         private const val DATABASE_NAME: String = "expense_db"
-        private lateinit var INSTANCE: ExpenseDatabase
-        public fun build(context: Context): ExpenseDatabase {
-            synchronized(ExpenseDatabase::class.java) {
-                if (!::INSTANCE.isInitialized) {
-                    INSTANCE = Room.databaseBuilder(
-                        context.applicationContext,
-                        ExpenseDatabase::class.java,
-                        DATABASE_NAME
-                    ).build()
-                }
-            }
-            return INSTANCE
-        }
+        public fun build(context: Context): ExpenseDatabase = Room.databaseBuilder(
+            context.applicationContext,
```

Comment:
Nice. Why's the function public.?

Reply:
I felt the class has already 'controlled' the visibility and function won't be visible outside of the cache module. I was also not sure if Room needs it to be public for some reason

Reply:
OK. If a class is internal, then all it's properties should either be internal or private

Reply:
Okay, I think I will make it internal too